packages/std/src/math/signed_decimal_256.rs

```diff
@@ -0,0 +1,3226 @@
+            )
+            .unwrap(),
+            SignedDecimal256::MIN
+        );
+        assert_eq!(
+            SignedDecimal256::from_str("-1").unwrap(),
+            SignedDecimal256::negative_one()
+        );
+    }
+
+    #[test]
+    fn signed_decimal_256_from_str_errors_for_broken_whole_part() {
+        match SignedDecimal256::from_str("").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str(" ").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("-").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing whole"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+    }
+
+    #[test]
+    fn signed_decimal_256_from_str_errors_for_broken_fractinal_part() {
+        match SignedDecimal256::from_str("1.").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing fractional"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("1. ").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing fractional"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("1.e").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing fractional"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("1.2e3").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing fractional"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        match SignedDecimal256::from_str("1.-2").unwrap_err() {
+            StdError::GenericErr { msg, .. } => assert_eq!(msg, "Error parsing fractional"),
+            e => panic!("Unexpected error: {e:?}"),
+        }
+    }
+
+    #[test]
+    fn signed_decimal_256_from_str_errors_for_more_than_18_fractional_digits() {
+        match SignedDecimal256::from_str("7.1234567890123456789").unwrap_err() {
+            StdError::GenericErr { msg, .. } => {
+                assert_eq!(msg, "Cannot parse more than 18 fractional digits",)
+            }
+            e => panic!("Unexpected error: {e:?}"),
+        }
+
+        // No special rules for trailing zeros. This could be changed but adds gas cost for the happy path.
+        match SignedDecimal256::from_str("7.1230000000000000000").unwrap_err() {
+            StdError::GenericErr { msg, .. } => {
+                assert_eq!(msg, "Cannot parse more than 18 fractional digits")
+            }
+            e => panic!("Unexpected error: {e:?}"),
+        }
+    }
```

Comment:
Similarily:
```
    #[test]
    fn signed_decimal_256_from_str_errors_for_more_than_18_fractional_digits() {
        let expected_err = StdError::generic_err("Cannot parse more than 18 fractional digits");
        assert_eq!(expected_err, SignedDecimal256::from_str("7.1234567890123456789").unwrap_err());
        // No special rules for trailing zeros. This could be changed but adds gas cost for the happy path.
        assert_eq!(expected_err, SignedDecimal256::from_str("7.1230000000000000000").unwrap_err());
    }
```